src/test/java/uk/gov/justice/digital/hmpps/vsip/pages/NeedAReviewPage.java

```diff
@@ -21,4 +21,15 @@ public void clickOnView()
     public void checkReviewMsg(String reviewMsg) {
         methodsService.isElementDisplayed("xpath","//h2[text()='Needs review']");
     }
+
+    public void checkNoReferenceIsDisplayed(String reference) {
```

Comment:
I worry that selectors like this with deep HTML structures (`.../div/div/table/tbody//td...`) are quite brittle: e.g. we - OR a GDS component change - might end up with a different nested structure of `div` elements but wouldn't expect that to break tests.

Is a more specific class/attribute needed on the target element?

Also, because this is testing that something **is not present** being so specific might make it falsely pass: e.g. we could be incorrectly displaying the reference number, just at a slightly different place in the DOM. Perhaps for a 'negative' test it's worth searching more broadly, e.g. in the entirety of `main-content` or the relevant table (which we can put a class/attrribute on) 🤷‍♂️

Reply:
I agree I'm going to re-visit this once Dhiraj finishes with API components that I need for execution bit. This is not a complete Test yet.

Reply:
👍 ok, well just let me or Ben know if you need extra selectors adding anywhere so we're directly targeting the right elements and not relying on HTML structure.

Reply:
yes sure Thanks Tom

Reply:
I will do my changes today and send it across. Might be easier that way instead of raising a new PR

Reply:
@anunagula3 - new Testing API helper endpoints added to add and remove exclude dates

Reply:
Got it. will be raising a separate PR for that.